### PR TITLE
Add test to verify docker images build

### PR DESCRIPTION
**Pull request by KieranHolroyd**
Currently fails on a number of the test cases, I'm not currently sure how to programatically fix this however, as it seems to be based on the underlying frameworks requiring certain files or folders to exist in order to complete the application build step. It might be worth considering pulling the latest version of whatever "starter project" these test candidates offer, removing any Dockerfile possibly shipped with the starter project, building it with the tool then verifying its correctness.

This pull request is not ready to be merged, I'm interested in feedback as to how this could be made to work easily with possible future test cases in mind.
(it seems my formatter changed a few things, which may go against code style, oops)

### CODE REVIEW

**Comment by rubys** [comment]
The "sister" project to this, [dockerfile-rails](https://github.com/rubys/dockerfile-rails) does start from scratch and uses rails to generate an application.  It even includes a single "system test" which launches an application runs the system tests contained within that application.  As that is slow, this is only done on one application.

**Comment by rubys** [comment]
I wonder if the docker setup build action would help?

https://github.com/rubys/dockerfile-rails/blob/main/.github/workflows/system-test.yml#L8-L8

**Comment by KieranHolroyd** [comment]
I certainly had some issues getting docker images to build on actions, but pending [this](https://github.com/KieranHolroyd/dockerfile-node/actions/runs/4833911287/jobs/8614499085) job not exiting early (the test will fail on 5 counts). I think it works properly

The next step would be to find some way of bootstrapping the projects to a buildable state for each test case, as this is currently not the case so the builds fail at the `npm run build` step or yarn/pnpm equivilant.

**Comment by KieranHolroyd** [comment]
Ok, so after some research I found [this issue](https://github.com/mochajs/mocha/issues/3559).
It turns out that the exit code 0-255 is the number of failing tests, I tested this and it is apparently the case.

This means that all the previous test cases failing with exit code 5, just have 5 failing tests which is correct, problem solved?

File: test/remix-indie/app/root.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      </head>
+      <body className="h-full">
+        <Outlet />
+        <ScrollRestoration />
+        <Scripts />
+        <LiveReload port={process.env.REMIX_DEV_SERVER_WS_PORT} />
+      </body>
+    </html>
+  );
+}
```
Comment (MichaelDeBoey) [inline test/remix-indie/app/root.tsx:65]: It seems that you're referencing a really old version of the `indie-stack`

```suggestion
import { cssBundleHref } from "@remix-run/css-bundle";
import type { LinksFunction, LoaderArgs } from "@remix-run/node";
import { json } from "@remix-run/node";
import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

import { getUser } from "~/session.server";
import tailwindStylesheetUrl from "~/styles/tailwind.css";

export const links: LinksFunction = () => [
  { rel: "stylesheet", href: tailwindStylesheetUrl },
  ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
];

export const loader = async ({ request }: LoaderArgs) => {
  return json({ user: await getUser(request) });
};

export default function App() {
  return (
    <html lang="en" className="h-full">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body className="h-full">
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}

```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: test/remix-indie/app/root.tsx (hunk 1 above)
Comment (rubys) [reply test/remix-indie/app/root.tsx:65]: @MichaelDeBoey you should be able to push this change directly to this branch.  Go for it!

**Comment by rubys** [comment]
Since all tests are passing, merge when you feel comfortable.

**Pull request merged**